codegen/masm/tests/test_basic_arithmetic.rs

```diff
@@ -0,0 +1,443 @@
+use air_codegen_masm::{code_gen, constants};
+use assembly::Assembler;
+use ir::AirIR;
+use processor::{
+    math::{Felt, FieldElement},
+    AdviceInputs, Kernel, MemAdviceProvider, Process, QuadExtension, StackInputs,
+};
+
+mod utils;
+use utils::{parse, to_stack_order};
+
+static ARITH_AIR: &str = "
+def SimpleArithmetic
+
+trace_columns:
+    main: [a, b]
+
+public_inputs:
+    stack_inputs: [16]
+
+boundary_constraints:
+    enf a.first = 0
+
+integrity_constraints:
+    enf a + a = 0
+    enf a - a = 0
+    enf a * a = 0
+
+    enf b + a = 0
+    enf b - a = 0
+    enf b * a = 0
+";
+
+#[test]
+fn test_simple_arithmetic() {
+    let ast = parse(ARITH_AIR);
+    let ir = AirIR::new(ast).expect("build AirIR failed");
+    let code = code_gen(&ir).expect("codegen failed");
+
+    let a = QuadExtension::new(Felt::new(3), Felt::ZERO);
+    let b = QuadExtension::new(Felt::new(7), Felt::ZERO);
+    let a_prime = a;
+    let b_prime = b;
+    let stack = to_stack_order(&[a, a_prime, b, b_prime]);
+    let code = format!(
+        "
+        begin
+            push.{}                                       # frame address
+            push.{}.{}.{}.{} dup.4 add.0 mem_storew dropw # save a and a'
+            push.{}.{}.{}.{} dup.4 add.1 mem_storew dropw # save b and b'
+            drop                                          # clean frame address
```

Comment:
Non-blocking comment: it might be nice to have a test util in the future to generate this.